genai-perf/tests/test_image_retrieval_profile_data_parser.py

```diff
@@ -138,3 +139,71 @@ def test_image_retrieval_profile_data(
 
         check_image_retrieval_metrics(metrics, expected_metrics)
         check_statistics(statistics, expected_statistics)
+
+    @patch("pathlib.Path.exists", return_value=True)
+    @patch(
+        "builtins.open",
+        new_callable=mock_open,
+        read_data=json.dumps(image_retrieval_profile_data),
+    )
+    @pytest.mark.parametrize(
+        "infer_mode, load_level, expected_metrics",
+        [
+            (
+                "concurrency",
+                "10",
+                {
+                    "request_throughputs": [1 / ns_to_sec(3)],
+                    "request_latencies": [2, 4],
+                    "image_throughputs": [1 / ns_to_sec(1), 3 / ns_to_sec(4)],
+                    "image_latencies": [1, 4 / 3],
+                    "request_goodputs": [1 / ns_to_sec(6)],
+                },
+            ),
+        ],
+    )
+    def test_image_retrieval_profile_data_with_goodput(
+        self,
+        mock_exists,
+        mock_file,
+        infer_mode,
+        load_level,
+        expected_metrics,
+    ) -> None:
+        """Collect image retrieval metrics from profile export data and check values.
+
+        Goodput Constraints
+        * image latency: 1.25e-6 ms
+        * request latency: 2.5e-6 ms
+
```

Comment:
We don't need to reiterate this information here as it's already in the code.

```suggestion
```

Reply:
Wouldn't this be more readable? As you can see clearly in the comment to see which request satisfy the constraints and hence quickly understand the process of computing goodput.

Reply:
I don't think this is about readability but more about are you giving any new information through docstring. For example, the docstring about metrics is useful because it shows you how the numbers are derived, instead of just reiterating the number. If you would like to make it more clear, then I suggest you add more information (e.g. how you are using goodput constraints to arrive at the final metric, etc).

Reply:
Updated, add computation process in the comment

Reply:
nice! thanks for adding the details